components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/autofill/LoginStorageDelegate.kt

```diff
@@ -0,0 +1,149 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.autofill
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import mozilla.components.lib.dataprotect.SecureAbove22Preferences
+import mozilla.components.service.sync.logins.AsyncLoginsStorage
+import mozilla.components.service.sync.logins.ServerPassword
+import org.mozilla.geckoview.GeckoResult
+import org.mozilla.geckoview.GeckoRuntime
+import org.mozilla.geckoview.LoginStorage
+import org.mozilla.geckoview.LoginStorage.LoginEntry
+
+/**
+ * Defines methods that will be implemented by [LoginStorage.Delegate] in future versions.
+ *
+ * TODO remove these once the GV API is complete.
```

Comment:
2/3 down, and iiuc `onLoginUsed` is not critical (it's used for telemetry, I think?).  I think we would be OK merging without it.

Reply:
Used for "last used" fields in the db (which are synced), but IIRC we're not really relying on these... maybe in the merge/dedupe logic during sync...?

Reply:
I think it's fine to defer `onLoginUsed` to a separate PR since the Gecko<=>GV side of it isn't implemented. Users can see the metadata on some platforms e.g. timeLastUsed on desktop but I think that doesn't need to block landing the doorhangers.